src/stores/DAOEntityStore.js

```diff
@@ -7,17 +7,26 @@ import { LoadObject } from 'brewskey.js-api';
 
 class DAOEntityStore<TEntity: { id: EntityID }> {
   _dao: DAO<TEntity, *>;
+  _queryOptions: QueryOptions = {};
 
   @observable _entityLoaders: Array<LoadObject<TEntity>> = [];
-  @observable _isFetching: boolean = false;
 
   constructor(dao: DAO<TEntity, *>) {
     this._dao = dao;
 
-    this._dao.subscribe(() => this._reload());
+    this._dao.subscribe(this._reload);
     this._reload();
   }
 
+  dispose = (): void => this._dao.unsubscribe(this._reload);
+
+  @action
+  setQueryOptions = (queryOptions: QueryOptions) => {
+    this._entityLoaders = [];
+    this._queryOptions = queryOptions;
+    this._reload();
```

Comment:
`_reload` should not be called here.  You should only call this when you use the `allItems` getter. That way you only fetch data when it's actually needed.